rust_snuba/src/processors/release_health_metrics.rs

```diff
@@ -0,0 +1,706 @@
+    timestamp: f64,
+    sentry_received_timestamp: Option<f64>,
+    tags: BTreeMap<String, u64>,
+    #[serde(flatten)]
+    value: MetricValue,
+    retention_days: u16,
+}
+
+fn default_use_case_id() -> String {
+    "sessions".into()
+}
+
+// #[derive(Debug, Deserialize)]
+// #[serde(untagged)]
+// enum MetricValue {
+//     Counter(f64),
+//     SetOrDistribution(Vec<u64>),
+//     DistributionFloat(Vec<f64>),
+// }
+
+#[derive(Debug, Deserialize)]
+#[serde(tag = "type", content = "value")]
+enum MetricValue {
+    #[serde(rename = "c")]
+    Counter(f64),
+    #[serde(rename = "s")]
+    Set(Vec<u64>),
+    #[serde(rename = "d")]
+    Distribution(Vec<f64>),
+}
+
+/// The raw row that is written to clickhouse.
+#[derive(Debug, Serialize, Default)]
+struct MetricsRawRow {
+    use_case_id: String,
+    org_id: u64,
+    project_id: u64,
+    metric_id: u64,
+    timestamp: u32,
+    retention_days: u16,
+    #[serde(rename = "tags.key")]
+    tags_key: Vec<u64>,
+    #[serde(default, rename = "tags.value")]
+    tags_value: Vec<u64>,
+    metric_type: String,
+    set_values: Option<Vec<u64>>,
+    count_value: Option<f64>,
+    distribution_values: Option<Vec<f64>>,
+    materialization_version: u8,
+    timeseries_id: u32,
+    partition: u16,
+    offset: u64,
+}
+
+/// Parse is the trait which should be implemented for all metric types.
+/// It is used to parse the incoming message into the appropriate raw row.
+/// Item represents the row into which the message should be parsed.
+trait Parse {
+    fn parse(
+        from: FromMetricsMessage,
+        meta: KafkaMessageMetadata,
+        config: &ProcessorConfig,
+    ) -> anyhow::Result<Option<MetricsRawRow>>;
+}
+
+impl Parse for MetricsRawRow {
+    fn parse(
+        from: FromMetricsMessage,
+        meta: KafkaMessageMetadata,
+        config: &ProcessorConfig,
+    ) -> anyhow::Result<Option<MetricsRawRow>> {
+        let timeseries_id =
+            generate_timeseries_id(from.org_id, from.project_id, from.metric_id, &from.tags);
+        let (tag_keys, tag_values): (Vec<_>, Vec<_>) = from.tags.into_iter().unzip();
+        let retention_days = enforce_retention(Some(from.retention_days), &config.env_config);
+
+        Ok(Some(MetricsRawRow {
+            use_case_id: from.use_case_id,
+            org_id: from.org_id,
+            project_id: from.project_id,
+            metric_id: from.metric_id,
+            timestamp: from.timestamp as u32,
+            retention_days,
+            tags_key: tag_keys.into_iter().map(|e| e.parse().unwrap()).collect(),
+            tags_value: tag_values,
+            materialization_version: 4,
+            timeseries_id,
+            partition: meta.partition,
+            offset: meta.offset,
+            ..match from.value {
+                MetricValue::Set(value) => MetricsRawRow {
+                    metric_type: "set".to_string(),
+                    set_values: Some(value),
+                    ..Default::default()
```

Comment:
Does this work by putting in the default values for the `count_value` and `distribution_values` fields? Or does it basically not have those fields in the new message output?

Reply:
The first, `Default::default()` puts `None` for `Option<T>`